### ISSUE TITLE
In-cluster build error with Kaniko

### ISSUE THREAD

**Issue by sk-credify** [garden-io/garden#2016]
### Current Behavior

Kaniko cannot push the image to the registry.

```
/.garden/comms/socatStarted
error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "127.0.0.1:5000/kaniko/frontend:v-cb3aca9dc9": POST http://127.0.0.1:5000/v2/kaniko/frontend/blobs/uploads/: unsupported status code 405; body: Method not allowed
```

https://i.imgur.com/aORhiz2.png

### Your environment
`garden version`
0.12.2

`kubectl version`
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-14T11:09:22Z", GoVersion:"go1.14.7", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.9-eks-4c6976", GitCommit:"4c6976793196d70bc5cd29d56ce5440c9473648e", GitTreeState:"clean", BuildDate:"2020-07-17T18:46:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

`docker version`

**Comment by edvald** [comment]
Sorry about the slow response. It appears you're running into this issue: #1996. The workaround there should help. We also have a fix for the cluster cleanup script coming up in the next release. Let us know if that works out.

**Comment by eysi09** [comment]
This was fixed with #2042.
